client/src/components/global/Date.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component, PropTypes } from 'react';
+import format from 'date-fns/format';
+import parse from 'date-fns/parse';
+
+export default class Date extends Component {
+  static propTypes = {
+    prefix: PropTypes.string,
+    format: PropTypes.string,
+    date: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.object
+    ])
+  };
+
+  formatDate() {
+    let date = null;
+    if (this.props.date) {
+      if (this.props.date instanceof Object) {
+        date = (
+          format(this.props.date, this.props.format)
+        );
+      } else {
+        date = (
+          format(parse(this.props.date), this.props.format)
+        );
+      }
+    }
+
+    return date;
+  }
+
+  formatString(date) {
+    let output = null;
+    if (this.props.prefix) {
+      output = this.props.prefix + ' ' + date;
+    } else {
+      output = date;
+    }
+
+    return output;
+  }
+
+  render() {
+    const date = this.formatDate();
+    return (
+      <span>{ this.formatString(date) }</span>
```

Comment:
Note that the above snippet implies one small change to your code, which is to pass the date prop value to this.formatDate rather than to access it from within the formatDate method. This is a subtle but significant change. My approach makes the code less stateful and more functional. We pull the part where the object state is accessed (`this.props.date`) into its own method (`value()`). This means that we could execute formatDate and formatString with a different date value, rather than the this.props.date value. This is important in react, because it's not uncommon to want to see what happens to a prop before it becomes `this.props` on the object (in lifecycle methods, for example). By limiting where you access component state and using a functional style, your code will require less refactoring when you run into those situations. When I say "functional style," I mean writing methods that do not access this.props directly, but rather have it passed to them as arguments—a functional method is one that will always return the same results with the same set of arguments. A stateful method is one that returns different results based on the state of the object. Does that kind of make sense?